### ISSUE TITLE
Sidebar Sub-Menu

### ISSUE THREAD

**Issue by MaxHillebrand** [zkSNACKs/WasabiDoc#39]
I'm a bit confused about the nuances of how to properly edit the sidebar menu. So far, I've edited the [`/docs/.vuepress/config.toml`](/docs/.vuepress/config.toml) but there are also [`config_json.js`](/docs/.vuepress/config_json.js) and [`config_yaml.yaml`](/docs/.vuepress/config_yaml.yaml), and it seems that the latter two are not updated. There should probably be a contributors guide in the [`WasabiDoc/README.md`](/README.md) with the details.

So far, the sidebar has hyperlinks to the markdown files, but it does not show the sub-headers within the file. For some [but not all] the files I think this is a good addition, for example the FAQ should show the questions in a dropdown. I don't really understand the configuration guide of the [VuePress Documentation on Sidebar](https://vuepress.vuejs.org/default-theme-config/#sidebar)...

Here the current `config.toml`:

```
    # FAQ Wasabi sidebar
    # ----------------------------------------
    [[themeConfig.sidebar."/FAQ/"]]
    title = ""
    collapsable = false
    children = [
        "/FAQ/",
        "/FAQ/FAQ-Introduction.md",
        "/FAQ/FAQ-Installation.md",
        "/FAQ/FAQ-UseWasabi.md",
        "/FAQ/FAQ-GeneralBitcoinPrivacy.md",
        "/FAQ/FAQ-Contribution.md",
    ]
```

**Comment by thunderBiscuit** [comment]
The way to do this is through the `sidebarDepth` option. You'll notice we have it set at `sidebarDepth = 0` on line 7 of the config file. This sets the behaviour for all files, but you can adjust/override this by section if you want. Depth of 0 means that no headers are expanded when you click on a file, whereas sidebarDepth = 1 means all headers level 1 will unroll and you can click to quickly get to them. 2 is the same, it will pull all header1 and header2 tags. 

Here is an example on the FAQ section:

```
    # FAQ Wasabi sidebar
    # ----------------------------------------
    [[themeConfig.sidebar."/FAQ/"]]
    title = ""
    sidebarDepth = 3
    collapsable = false
    children = [
        "/FAQ/",
        "/FAQ/FAQ-Introduction.md",
        "/FAQ/FAQ-Installation.md",
        "/FAQ/FAQ-UseWasabi.md",
        "/FAQ/FAQ-GeneralBitcoinPrivacy.md",
        "/FAQ/FAQ-Contribution.md",
    ]
```

I do not have an opinion on what value to choose, I just wanted to explain how it works. Have you tried a few @MaxHillebrand ? What do you think? I do like it for the FAQ potentially, depending on how long the list is haha.

**Comment by thunderBiscuit** [comment]
Also I just noticed that your link to the VuePress documentation is for vuepress 0.something. It is a bit older and does not contain all the explanations for the stuff we are implementing, so you might have not been able to find something you were looking for... Check out the version 1 instead. Honestly I don't know why they don't put a banner on the old version making sure users know that this is not the docs related to the current version, the one that downloads by default!

https://v1.vuepress.vuejs.org/guide/

**Comment by MaxHillebrand** [comment]
As always, fantastic explanation, thank you so much @thunderBiscuit !!

As I read it [here](https://v1.vuepress.vuejs.org/theme/default-theme-config.html#nested-header-links), the maximum depth is 2, and I think that is good for most pages. I'll set it manually for all of the four pillars so that it can be modified individually without much hassle.

**Comment by thunderBiscuit** [comment]
Yeah I think that's a better approach than what I had. Explicit is better than implicit. This will make it clear to newcomers to the file.

**Comment by MaxHillebrand** [comment]
This is implemented by 64bb9cdfe929388b18956b42034d61d606e22138.
